compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -50,32 +50,35 @@
   private final SparkContext sparkContext;
   private final Integer parallelism;
   private final Stack<LoopVertex> loopVertexStack;
-  private DAGBuilder<IRVertex, IREdge> builder;
+  private final DAG<IRVertex, IREdge> dag;
   private final IRVertex lastVertex;
   private final KryoSerializer kryoSerializer;
 
   /**
-   * Constructor to start with.
+   * Static method to create a JavaRDD object.
    * @param sparkContext spark context containing configurations.
    * @param parallelism parallelism information.
+   * @param <T> type of the resulting object.
+   * @return the new JavaRDD object.
    */
-  JavaRDD(final SparkContext sparkContext, final Integer parallelism) {
-    this(sparkContext, parallelism, new DAGBuilder<>(), null);
+  static <T extends Serializable> JavaRDD<T> of(final SparkContext sparkContext, final Integer parallelism) {
+    return new JavaRDD<>(sparkContext, parallelism,
+        new DAGBuilder<IRVertex, IREdge>().buildWithoutSourceSinkCheck(), null);
   }
 
   /**
    * Constructor.
    * @param sparkContext spark context containing configurations.
    * @param parallelism parallelism information.
-   * @param builder the builder for the DAG.
+   * @param dag the current DAG.
    * @param lastVertex last vertex added to the builder.
    */
   private JavaRDD(final SparkContext sparkContext, final Integer parallelism,
-                  final DAGBuilder<IRVertex, IREdge> builder, final IRVertex lastVertex) {
+                  final DAG<IRVertex, IREdge> dag, final IRVertex lastVertex) {
```

Comment:
`@Nuallable final IRVertex lastVertex` ?

Reply:
Thanks for the suggestion!